code/game/objects/effects/decals/Cleanable/humans.dm

```diff
@@ -57,6 +94,87 @@ GLOBAL_LIST_EMPTY(splatter_cache)
 	color = adjust_brightness(color, -50)
 	amount = 0
 
+/obj/effect/decal/cleanable/blood/Crossed(atom/movable/O)
+	. = ..()
+	if(ishuman(O) && !gravity_check)
+		var/mob/living/carbon/human/H = O
+		var/list/obj/item/things_to_potentially_bloody = list()
+		var/count = amount + 1
+
+		for(var/obj/item/i in H.contents)
+			things_to_potentially_bloody |= i
+
+		if(length(things_to_potentially_bloody))
+			for(var/i in 1 to count)
+				things_to_potentially_bloody[rand(1, length(things_to_potentially_bloody))].add_blood(blood_DNA, basecolor)
+				count--
+				if(!count)
+					break
+		else
+			splat(get_turf(O))
```

Comment:
This needs to return because splat already qdels the blood at the end of the proc. We don't want to double qdel.
```suggestion
			splat(get_turf(O))
			return
```